Runtime/AvatarCreator/Scripts/Managers/AvatarManager.cs

```diff
@@ -288,9 +290,32 @@ public async Task<GameObject> UpdateAsset(AssetType assetType, BodyType bodyType
             {
                 return null;
             }
+            
+#if UNITY_EDITOR
+            await ValidateAssetUpdate(assetType, assetId);
+#endif
 
             return await inCreatorAvatarLoader.Load(avatarId, bodyType, gender, data);
         }
+        
+#if UNITY_EDITOR
+        private async Task ValidateAssetUpdate(AssetType assetType, object assetId)
+        {
+            if (assetType != AssetType.BodyShape)
```

Comment:
because of this check here, probably you should rename it to something like ValidateBodyShapes or CheckForBodyshapes.
However if you want to keep it as a more generic Validate asset check that can be used on any asset type, then I would remove this if statement and instead wrap the calling of this function in it.  
EG  
```cs
if(assetType == AssetType.BodyShape){
await ValidateAssetUpdate(assetType, assetId);
}
```

Although if you do this you might need to make the error message more dynamic, like "failed to apply {assetType}..." etc.

Reply:
good point